network/ovssnat/ovssnat.go

```diff
@@ -311,26 +314,58 @@ func (client *OVSSnatClient) DeleteSnatEndpoint() error {
 	return nil
 }
 
+func setBridgeMac(hostPrimaryMac string) error {
+	hwAddr, err := net.ParseMAC(hostPrimaryMac)
+	if err != nil {
+		log.Errorf("Error while parsing host primary mac: %s error:%+v", hostPrimaryMac, err)
+		return err
+	}
+
+	if err = netlink.SetLinkAddress(SnatBridgeName, hwAddr); err != nil {
+		log.Errorf("Error while setting macaddr on bridge: %s error:%+v", hwAddr.String(), err)
+	}
+	return err
+}
+
+func dropArpForSnatBridgeIP(snatBridgeIP, azSnatVethIfName string) error {
+	var err error
+	_, ipCidr, _ := net.ParseCIDR(snatBridgeIP)
+	if err = ebtables.SetArpDropRuleForIp(ipCidr.String(), azSnatVethIfName); err != nil {
+		log.Errorf("Error setting arp drop rule for snatbridge ip :%s", snatBridgeIP)
+	}
+
+	return err
+}
+
 /**
 	This function creates linux bridge which will be used for outbound connectivity by NCs
 **/
-func CreateSnatBridge(snatBridgeIP string, mainInterface string) error {
+func CreateSnatBridge(snatBridgeIP string, hostPrimaryMac string, mainInterface string) error {
 	_, err := net.InterfaceByName(SnatBridgeName)
 	if err == nil {
 		log.Printf("Snat Bridge already exists")
-		return nil
-	}
+	} else {
+		log.Printf("[net] Creating Snat bridge %v.", SnatBridgeName)
+
+		link := netlink.BridgeLink{
+			LinkInfo: netlink.LinkInfo{
+				Type: netlink.LINK_TYPE_BRIDGE,
+				Name: SnatBridgeName,
+			},
+		}
 
-	log.Printf("[net] Creating Snat bridge %v.", SnatBridgeName)
+		if err := netlink.AddLink(&link); err != nil {
+			return err
+		}
+	}
 
-	link := netlink.BridgeLink{
-		LinkInfo: netlink.LinkInfo{
-			Type: netlink.LINK_TYPE_BRIDGE,
-			Name: SnatBridgeName,
-		},
+	log.Printf("Setting snat bridge mac: %s", hostPrimaryMac)
+	if err := setBridgeMac(hostPrimaryMac); err != nil {
+		return err
 	}
 
-	if err := netlink.AddLink(&link); err != nil {
+	log.Printf("Drop ARP for snat bridge ip: %s", snatBridgeIP)
+	if err := dropArpForSnatBridgeIP(snatBridgeIP, azureSnatVeth0); err != nil {
 		return err
 	}
```

Comment:
are these calls idempotent? previously we were just returning if snatbridge exsited?

Reply:
@chandanAggarwal  yes these calls are idempotent...for ebtable we check if rule exists and add it..the other one where we set mac to bridge is idempotent..i tested this in runner setup which creates multuple swift pods and each pod invoke this codepath